source/Set.h

```diff
@@ -34,12 +34,18 @@ class Set {
 
 	bool Has(const std::string &name) const { return data.count(name); }
 
-	typename std::map<std::string, Type>::iterator begin() { return data.begin(); }
-	typename std::map<std::string, Type>::const_iterator begin() const { return data.begin(); }
-	typename std::map<std::string, Type>::iterator end() { return data.end(); }
-	typename std::map<std::string, Type>::const_iterator end() const { return data.end(); }
+	using setIt = typename std::map<std::string, Type>::iterator;
+	using setCIt = typename std::map<std::string, Type>::const_iterator;
+
+	setIt begin() { return data.begin(); }
+	setCIt begin() const { return data.begin(); }
+	setIt find(const std::string &key) { return data.find(key); }
+	setCIt find(const std::string &key) const { return data.find(key); }
+	setIt end() { return data.end(); }
+	setCIt end() const { return data.end(); }
```

Comment:
There's no reason to provide the non-const version of `find`.

```suggestion
	using std::map<std::string, Type>::iterator;
	using std::map<std::string, Type>::const_iterator;

	iterator begin() noexcept { return data.begin(); }
	const_iterator begin() const noexcept { return data.begin(); }
	iterator end() noexcept { return data.end(); }
	const_iterator end() const noexcept { return data.end(); }
	const_iterator find(const std::string &key) const { return data.find(key); }
```
(I believe the using declarations here are valid--i.e. we don't have to alias the longer name, we can just bring it into the local scope. It's possible that is incorrect, and one should verify neither the names nor the aliases (if used instead) are brought into scope in including classes.)

Reply:
g++ didn't accept this construct, but `using` in a header file is a bit tricky, so I reverted back to the original code where there was no using statement.